SSA/Projects/InstCombine/LLVM/Semantics.lean

```diff
@@ -21,6 +21,12 @@ namespace IntW
 
 instance : Inhabited (IntW w) := by unfold IntW; infer_instance
 
+instance : Refinement (LLVM.IntW w) := inferInstanceAs (Refinement <| PoisonOr _)
+@[simp, simp_llvm]
+theorem isRefinedBy_iff (x y : LLVM.IntW w) :
+    x ⊑ y ↔ @HRefinement.IsRefinedBy (PoisonOr _) (PoisonOr _) _ x y := by
+  rfl
```

Comment:
It's not clear to me why this is the "good" direction of the simplification? In general, the typeclass tower is unclear to me!

Reply:
I added a docstring that hopefully clarifies:
```
/--
`isRefinedBy_iff` rewrites refinement of `LLVM.IntW` values into refinement
of `PoisonOr _` values, effectively unfolding the `LLVM.IntW` definition (and
the Refinement instance).

By making this a simp-lemma, we ensure all `PoisonOr` simp-lemmas apply without
having to duplicate that API for `LLVM.IntW`.
-/
```

I'd agree that normally this rewrite might not want to a `simp` rewrite, as it effectively means that the simp-normal form for `LLVM.IntW` becomes `PoisonOr`. However, given that we were originally OK with the former being an abbrev, in this case the rewrite seems fine to me. Certainly, duplicating all PoisonOr simp-lemmas does not seem worth the maintenance hassle to me.